@here/harp-mapview/lib/geometry/TileGeometryCreator.ts

```diff
@@ -711,55 +711,56 @@ export class TileGeometryCreator {
                 }
 
                 if (isLineTechnique(technique) || isSegmentsTechnique(technique)) {
-                    const hasDynamicColor =
-                        Expr.isExpr(technique.color) || Expr.isExpr(technique.opacity);
                     const fadingParams = this.getFadingParams(displayZoomLevel, technique);
                     FadingFeature.addRenderHelper(
                         object,
                         viewRanges,
                         fadingParams.fadeNear,
-                        fadingParams.fadeFar,
-                        false,
-                        hasDynamicColor
-                            ? (renderer, mat) => {
-                                  const lineMaterial = mat as THREE.LineBasicMaterial;
-                                  applyBaseColorToMaterial(
-                                      lineMaterial,
-                                      lineMaterial.color,
-                                      technique,
-                                      technique.color,
-                                      mapView.zoomLevel
-                                  );
-                              }
-                            : undefined
+                        fadingParams.fadeFar
                     );
+
+                    if (Expr.isExpr(technique.color) || Expr.isExpr(technique.opacity)) {
+                        tile.addUpdater(() => {
```

Comment:
As far as I can see we still register one callback per object, even if the material is the same for all objects.
I would prefer to have a function like `Tile.updateMaterials` and that one is explicitly iterating over the materials/techniques and updates them.

Reply:
Agree, my bad i didn't noticed that we always call update per object, not material even if only one update is needed.
Will fix.

Reply:
I tried to be generic, as these updaters are meant to update not only materials, but also 
- `TextElement`s - at least `priority`, `color`, `size` has to be interpolated and it was reported as bug.
-  possibly object properties (like enable/disable depth pre-pass or even remove edge/outline objects if they become unnecessary when their size/width interpolate to 0 or null) 

In next followup PoC PR, i'll show my idea, in this one i'll try to focus on moving updates to controlled place (outside of onBeforeRender), so `object.material.opacity` & `textElement.renderStyle,opacity` is always updated in controlled manner.